cli: TestDemoLocality failed [failed connection attempt; no certificates found]

cli.TestDemoLocality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20591774?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20591774?buildTab=artifacts#/) on master @ [c3417908798c54be899e0b9af748c2e568d83627](https://github.com/cockroachdb/cockroach/commits/c3417908798c54be899e0b9af748c2e568d83627):


```
sync.runtime_notifyListWait(0x4001f96a50, 0x17b)
	GOROOT/src/runtime/sema.go:587 +0x154
sync.(*Cond).Wait(0x4001f96a40)
	GOROOT/src/sync/cond.go:71 +0xcc
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftSchedulerShard).worker(0x400c8df2d0, {0x7500c08, 0x40047152c0}, {0x752d660, 0x4005310008}, 0x4007e06008)
	pkg/kv/kvserver/scheduler.go:378 +0x368
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start.func2({0x7500c08, 0x40047152c0}, 0x7500c08?)
	pkg/kv/kvserver/scheduler.go:314 +0xe0
created by github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*raftScheduler).Start in goroutine 7534
	pkg/kv/kvserver/scheduler.go:328 +0x168

goroutine 12004 [select, 14 minutes]:
google.golang.org/grpc/internal/grpcsync.(*CallbackSerializer).run(0x4002d00640, {0x74ff2b8, 0x400a13e910})
	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:83 +0xc4
created by google.golang.org/grpc/internal/grpcsync.NewCallbackSerializer in goroutine 12003
	external/org_golang_google_grpc/internal/grpcsync/callback_serializer.go:55 +0x118

goroutine 8330 [select, 14 minutes]:
github.com/cockroachdb/cockroach/pkg/sql.(*runnerCoordinator).init.func1({0x4006db6240?, 0x7500c08?})
	pkg/sql/distsql_running.go:169 +0xb0
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x7500c08, 0x40047152c0})
	pkg/util/stop/stopper.go:440 +0x88
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 6415
	pkg/util/stop/stopper.go:438 +0x104

goroutine 10499 [select, 14 minutes]:
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.(*UnbufferedSender).run(0x4004019a00, {0x74ff2b8, 0x400581d180}, 0x4003ccc090, 0x4002672090)
	pkg/kv/kvserver/rangefeed/unbuffered_sender.go:153 +0xa0
github.com/cockroachdb/cockroach/pkg/kv/kvserver/rangefeed.(*StreamManager).Start.func1({0x74ff2b8, 0x400581d180})
	pkg/kv/kvserver/rangefeed/stream_manager.go:192 +0xc0
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x74ff2b8, 0x400581d180})
	pkg/util/stop/stopper.go:440 +0x88
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 10498
	pkg/util/stop/stopper.go:438 +0x104

goroutine 8960 [select]:
github.com/cockroachdb/cockroach/pkg/sql.(*runnerCoordinator).init.func1({0x40019c9e40?, 0x7500c08?})
	pkg/sql/distsql_running.go:169 +0xb0
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x7500c08, 0x400a1a0640})
	pkg/util/stop/stopper.go:440 +0x88
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 8883
	pkg/util/stop/stopper.go:438 +0x104

goroutine 8942 [select, 10 minutes]:
github.com/cockroachdb/cockroach/pkg/sql.(*runnerCoordinator).init.func1({0x4001ac0480?, 0x7500c08?})
	pkg/sql/distsql_running.go:169 +0xb0
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x7500c08, 0x400a1a0640})
	pkg/util/stop/stopper.go:440 +0x88
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 8883
	pkg/util/stop/stopper.go:438 +0x104
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations @cockroachdb/server @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDemoLocality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55340

cli.TestDemoLocality [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20634754?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20634754?buildTab=artifacts#/) on master @ [bf25721cad9935027bbe409460e88c380a7c2b4a](https://github.com/cockroachdb/cockroach/commits/bf25721cad9935027bbe409460e88c380a7c2b4a):


```
W251022 16:42:41.308744 33048 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T1,Vsystem,n1,rangefeed=system-config-cache,mux_n=1,gen=2] 2041  error closing mux rangefeed client: context canceled
W251022 16:42:41.308820 33527 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T1,Vsystem,start-demo-cluster,phase=2,start-n1,n1,rangefeed=tenant-boundaries-watcher,mux_n=1,gen=2] 2042  error closing mux rangefeed client: context canceled
I251022 16:42:41.308857 33470 sql/stats/automatic_stats.go:771  [T1,Vsystem,n1] 2043  quiescing stats garbage collector
W251022 16:42:41.308713 32369 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T1,Vsystem,n1,rangefeed=sql_instances,mux_n=1,gen=2] 2040  error closing mux rangefeed client: context canceled
W251022 16:42:41.308893 33631 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T1,Vsystem,n1,rangefeed=tenant-settings-watcher,mux_n=1,gen=2] 2044  error closing mux rangefeed client: context canceled
W251022 16:42:41.308914 33170 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T1,Vsystem,n1,job=AUTO SPAN CONFIG RECONCILIATION id=1117654088486584321,rangefeed=sql-watcher-zones-rangefeed,mux_n=1,gen=2] 2045  error closing mux rangefeed client: context canceled
I251022 16:42:41.308995 32998 jobs/registry.go:1568  [T1,Vsystem,n1,job=1117654088486584321] 2046  AUTO SPAN CONFIG RECONCILIATION job 1117654088486584321: stepping through state succeeded
E251022 16:42:41.309188 32998 jobs/registry.go:918  [T1,Vsystem,n1] 2047  error getting live session: node unavailable; try another peer
W251022 16:42:41.308773 31733 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T1,Vsystem,n1,rangefeed=lease,mux_n=1,gen=2] 2048  error closing mux rangefeed client: context canceled
W251022 16:42:41.308851 33622 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T1,Vsystem,n1,rangefeed=statement-hints-watcher,mux_n=1,gen=2] 2049  error closing mux rangefeed client: context canceled
W251022 16:42:41.308859 33654 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T1,Vsystem,start-demo-cluster,phase=2,start-n1,n1,rangefeed=tenant-entry-watcher,mux_n=1,gen=2] 2050  error closing mux rangefeed client: context canceled
W251022 16:42:41.308899 33419 15@kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:407  [T1,Vsystem,n1,rangefeed=table-stats-cache,mux_n=1,gen=2] 2051  error closing mux rangefeed client: context canceled
I251022 16:42:41.361580 28275 13@kv/kvserver/allocator/plan/replicate.go:421  [T1,Vsystem,n1,replicate,s1,r4/1:/System{/tsd-tse}] 2052  adding voter n5,s5: [1*:38]
I251022 16:42:41.362514 28275 15@kv/kvserver/replica_command.go:1923  [T1,Vsystem,n1,replicate,s1,r4/1:/System{/tsd-tse}] 2053  could not successfully add and upreplicate LEARNER replica(s) on [n5,s5], rolling back: change replicas of r4 failed: fetching current range descriptor value: aborted in DistSender: result is ambiguous: node unavailable; try another peer
I251022 16:42:41.362605 28275 13@kv/kvserver/replicate_queue.go:859  [T1,Vsystem,n1,replicate,s1,r4/1:/System{/tsd-tse}] 2054  error processing replica: change replicas of r4 failed: fetching current range descriptor value: aborted in DistSender: result is ambiguous: node unavailable; try another peer
W251022 16:42:41.362695 28275 13@kv/kvserver/queue.go:1478  [T1,Vsystem,n1,replicate,s1,r4/1:/System{/tsd-tse}] 2055  queue processing resulted in non-benign error: change replicas of r4 failed: fetching current range descriptor value: aborted in DistSender: result is ambiguous: node unavailable; try another peer
I251022 16:42:41.362760 28275 13@kv/kvserver/queue.go:1629  [T1,Vsystem,n1,replicate] 2056  purgatory is now empty
W251022 16:42:41.362780 31755 13@kv/kvserver/queue.go:1478  [T1,Vsystem,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 2057  queue processing resulted in non-benign error: node unavailable; try another peer
W251022 16:42:41.362806 31755 13@kv/kvserver/queue.go:1204  [T1,Vsystem,n1,replicate,s1,r1/1:/{Min-System/NodeL…}] 2058  queue.replicate: processing replica [outer]: task did not start node unavailable; try another peer
E251022 16:42:41.587390 30828 2@rpc/peer.go:673  [testclient,user=root,nsql=1,rnode=1,raddr=127.0.0.1:45109,class=default,rpc] 2059  disconnected (was healthy for 2.001s): grpc: node unavailable; try another peer [code 2/Unknown]
E251022 16:42:41.588055 30828 2@rpc/peer.go:694  [testclient,user=root,nsql=1,rnode=1,raddr=127.0.0.1:45109,class=default,rpc] 2060  failed connection attempt (last connected 1ms ago): grpc: connection error: desc = "transport: error while dialing: dial tcp 127.0.0.1:45109: connect: connection refused" [code 14/Unavailable]
W251022 16:42:42.257987 8062 server/server_sql.go:1812  [T1,Vsystem,n1] 2061  server shutdown without a prior graceful drain
I251022 16:42:42.263501 8062 server/server_controller.go:416  [T1,Vsystem,n1] 2062  server controller shutting down
I251022 16:42:42.263541 8062 server/server_controller.go:425  [T1,Vsystem,n1] 2063  waiting for tenant servers to report stopped
I251022 16:42:42.279835 8062 cli/testutils.go:297  [-] 2064  stopping server and cleaning up CLI test
    demo_locality_test.go:41: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7574/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/cli/cli_test_/cli_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/testdata/demo/test_demo_locality:11:
          exec
          demo
          --nodes=9
          --execute=select node_id, locality from crdb_internal.gossip_nodes order by node_id
        output didn't match expected:
          @@ -1,11 +1,5 @@
          -node_id	locality
          -1	region=us-east1,az=b
          -2	region=us-east1,az=c
          -3	region=us-east1,az=d
          -4	region=us-west1,az=a
          -5	region=us-west1,az=b
          -6	region=us-west1,az=c
          -7	region=europe-west1,az=b
          -8	region=europe-west1,az=c
          -9	region=europe-west1,az=d
          +ERROR: cannot load certificates.
          +Check your certificate settings, set --certs-dir, or use --insecure for insecure clusters.
           
          +initial connection heartbeat failed: problem using security settings: no certificates found; does certs dir exist?
          +
    testutils.go:290: -- test log scope end --
--- FAIL: TestDemoLocality (13.60s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDemoLocality.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/security/securityassets/security_assets.go
+++ b/pkg/security/securityassets/security_assets.go
@@ -69,6 +69,12 @@ func readDir(dirname string) ([]os.FileInfo, error) {
 	for _, entry := range entries {
 		info, err := entry.Info()
 		if err != nil {
+			// Skip files that disappeared between ReadDir and Info().
+			// This can happen when the directory contains transient files
+			// like Unix socket lock files that are created/deleted rapidly.
+			if oserror.IsNotExist(err) {
+				continue
+			}
 			return nil, err
 		}
 		infos = append(infos, info)

